erpnext/selling/page/point_of_sale/point_of_sale.py

```diff
@@ -139,8 +139,16 @@ def get_conditions(item_code, serial_no, batch_no, barcode):
 	if serial_no or batch_no or barcode:
 		return "item.name = {0}".format(frappe.db.escape(item_code))
 
-	return """(item.name like {item_code}
-		or item.item_name like {item_code})""".format(item_code = frappe.db.escape('%' + item_code + '%'))
+	cond = "("
+	cond += """item.name like {item_code}
+		or item.item_name like {item_code}""".format(item_code = frappe.db.escape('%' + item_code + '%'))
+	search_fields = frappe.get_all('POS Search Fields', fields = ['fieldname'])
+	if search_fields:
+		for field in search_fields:
```

Comment:
We should probably verify if the field is not already used in the default condition. For eg. item_name

If such is the case, do not add the custom search condition. 

Or we could stop them from getting added to the child table itself by a python validation on POS settings.

Reply:
The better approach in my opinion is to not show those in the drop-down itself. Hence it won't be available for selection and solve our issue. What do you say?